hbase-client/src/main/java/org/apache/hadoop/hbase/client/HBaseAdmin.java

```diff
@@ -1921,6 +1922,34 @@ protected NamespaceDescriptor rpcCall() throws Exception {
     });
   }
 
+  /**
+   * List available namespaces
+   * @return List of namespace names
+   * @throws IOException if a remote or network exception occurs
+   */
+  @Override
+  public String[] listNamespaces() throws IOException {
+    return executeCallable(new MasterCallable<String[]>(getConnection(),
+        getRpcControllerFactory()) {
+      @Override
+      protected String[] rpcCall() throws Exception {
+        List<String> list =
+          master.listNamespaces(getRpcController(),
+            ListNamespacesRequest.newBuilder().build()).getNamespaceNameList();
+        String[] res = new String[list.size()];
+        for(int i = 0; i < list.size(); i++) {
+          res[i] = list.get(i);
+        }
```

Comment:
Nit:
should we just use toArray to convert a list to an array, removing the for loop?
res = list.toArray(res);

Reply:
Ok, will fix on commit